geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -701,17 +701,17 @@ public boolean put(Object object) throws InterruptedException, CacheException {
     AbstractBucketRegionQueue brq =
         (AbstractBucketRegionQueue) prQ.getDataStore().getLocalBucketById(bucketId);
 
+    // Full path of the bucket:
+    final String bucketFullPath =
```

Comment:
Pulling this out will always construct this string now.  Any ideas on perf impact?

Reply:
Good point, I haven't ran any performance tests related to the change.
I've refactored the first part of the `String` as a constant and moved the initialisation _**only**_ to places where is needed, hopefully this will reduce the impact. Let me know if you think that's enough.